lemmy.yml

```diff
@@ -61,6 +61,11 @@
         owner: '991'
 
   - block:
+    - set_fact:
+        lemmy_port: "{{ 32767 |random(start=1024) }}"
+        lemmy_ui_port: "{{ 32767 |random(start=1024) }}"
+        pictrs_port: "{{ 32767 |random(start=1024) }}"
```

Comment:
This will generate a new, random port on each run. I believe this could cause errors if docker-compose already uses the new port, but nginx still tries to connect to the old one. Plus it will cause unnecessary changes in diffs. Besides, if the generated port is already taken, it will cause errors.

So i think it would be best to generate the ports only on first run, and then store them in a file for subsequent runs, similar to the [password lookup](https://github.com/LemmyNet/lemmy-ansible/blob/main/lemmy.yml#L90). Unfortunately i cant find an easy way to do that.

Reply:
The nginx is also overwritten and reloaded, so that should't be a problem.

Reply:
That is a separate step, so there will be some time between docker port changing, and nginx port changing. On a big instance that will definitely affect some users. And it will probably also break websocket connections.

Reply:
It pry wouldn't take more than 20 seconds. I'll test websockets right now.